packages/cardano-services/test/Asset/CardanoTokenRegistry.test.ts

```diff
@@ -0,0 +1,74 @@
+import { Asset, Cardano } from '@cardano-sdk/core';
+import { CardanoTokenRegistry } from '../../src/Asset';
+import { InMemoryCache, Key } from '../../src/InMemoryCache';
+import { dummyLogger } from 'ts-log';
+
+describe('CardanoTokenRegistry', () => {
+  const nonvalidAssetId = Cardano.AssetId('0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef');
+  const validAssetId = Cardano.AssetId('f43a62fdc3965df486de8a0d32fe800963589c41b38946602a0dc53541474958');
+
+  describe('return value', () => {
+    const tokenRegistry = new CardanoTokenRegistry({ logger: dummyLogger });
+
+    afterAll(() => tokenRegistry.shutdown());
+
+    it('returns null for nonexisting AssetId', async () => {
+      expect(await tokenRegistry.getTokenMetadata([nonvalidAssetId])).toEqual([null]);
+    });
+
+    it('returns metadata when subject exists', async () => {
+      const [metadata] = await tokenRegistry.getTokenMetadata([validAssetId]);
+
+      expect(metadata).not.toBeNull();
+      if (!metadata) return;
```

Comment:
assertion will interrupt the test when something is wrong

Reply:
Just to give you the reason of line 23.

The only meaning of line 23 is because it is partes by TypeScript while line 22 is not: till line 23 (included) `metadata` is of type `Asset.TokenMetadata | null` while from line 24 on `metadata` is of type `Asset.TokenMetadata`.

Once said that I can remove line 23 and use the exclamation mark operator after `metadata`... BTW it is used only once later...

Reply:
That was ...   unexpected 😀 
I'd vote for exclamation mark usage.
 

Reply:
Since both @rhyslbw and @iadmytro are still asking for clarifications about this, it is clear I have to apologize for my bad/wrong communication.

Please let me recap: I agree removing line 23 and using the question mark operator, I already applied the change but I still need tu push it.